test/service/AppealStorageService.test.ts

```diff
@@ -0,0 +1,71 @@
+import { AxiosResponse } from 'axios';
+import { expect } from 'chai';
+import { CREATED } from 'http-status-codes';
+import nock = require('nock');
+
+import { Appeal } from 'app/models/Appeal';
+import { AppealStorageService } from 'app/service/AppealStorageService'
+
+const appeal: Appeal = {
+    penaltyIdentifier: {
+        companyNumber: '00345567',
+        penaltyReference: 'A00000001',
+    },
+    reasons: {
+        other: {
+            title: 'I have reasons',
+            description: 'they are legit'
+        }
+    }
+};
+
+describe('AppealStorageService', () => {
+
+    describe('saving appeals', () => {
+
+        it('should throw an error when token not defined', () => {
+            const appealStorageService = new AppealStorageService('/companies/1/appeals');
+
+            [undefined, null].forEach(async invalidToken => {
+                try {
+                    await appealStorageService.save(appeal, invalidToken as any)
+                } catch (err) {
+                    expect(err).to.be.instanceOf(Error)
+                        .and.to.haveOwnProperty('message').equal('Token is missing')
+                }
+            })
+        });
+
+        it('should save appeal and status 201 with location header', async () => {
+
+            const appealStorageService = new AppealStorageService('http://localhost:9000');
+
+            nock('http://localhost:9000')
+                .post('/companies/00345567/appeals', {
+                        penaltyIdentifier: {
+                            companyNumber: '00345567',
+                            penaltyReference: 'A00000001',
+                        },
+                        reasons: {
+                            other: {
+                                title: 'I have reasons',
+                                description: 'they are legit'
+                            }
+                        }
+                    }, {
+                        reqheaders: {
+                            authorization: 'Bearer 123',
+                        },
+                    }
+                )
+                .reply(201, {}, {'location': '/companies/00345567/appeals/555'});
```

Comment:
Please test other service responses such as 400, 401 and 500.

Reply:
Covered all except from a 400 response code as the function validates the token/appeal parameters before passing to the api. API only returns 400 if the data is not sent, but a 422 if data is invalid (which i've covered)